k8s/operator.go

```diff
@@ -0,0 +1,127 @@
+package k8s
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	tykModePro = "pro"
+	tykModeKey = "TYK_MODE"
+	tykAuthKey = "TYK_AUTH"
+	tykOrgKey  = "TYK_ORG"
+	tykURLKey  = "TYK_URL"
+)
+
+// BootstrapTykOperatorSecret bootstrap a Kubernetes Secret utilized by Tyk Operator.
+// If the system has the secret created already, it deletes the existing one and recreates
+// a secret for Tyk Operator.
+func (c *Client) BootstrapTykOperatorSecret() error {
+	secrets, err := c.clientSet.
+		CoreV1().
+		Secrets(c.appArgs.K8s.ReleaseNamespace).
+		List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for i := range secrets.Items {
+		secret := secrets.Items[i]
+		if secret.Name == c.appArgs.OperatorKubernetesSecretName {
+			err = c.clientSet.
+				CoreV1().
+				Secrets(c.appArgs.K8s.ReleaseNamespace).
+				Delete(context.TODO(), secret.Name, metav1.DeleteOptions{})
+
+			if err != nil {
+				return err
+			}
+
+			c.l.Info("A previously created operator secret was identified and deleted")
+
+			break
+		}
+	}
+
+	secretData := map[string][]byte{
+		tykAuthKey: []byte(c.appArgs.Tyk.Admin.Auth),
+		tykOrgKey:  []byte(c.appArgs.Tyk.Org.ID),
+		tykModeKey: []byte(tykModePro),
+		tykURLKey:  []byte(c.appArgs.K8s.DashboardSvcUrl),
+	}
+
+	objectMeta := metav1.ObjectMeta{Name: c.appArgs.OperatorKubernetesSecretName}
+
+	secret := v1.Secret{
+		ObjectMeta: objectMeta,
+		Data:       secretData,
+	}
+
+	_, err = c.clientSet.
+		CoreV1().
+		Secrets(c.appArgs.K8s.ReleaseNamespace).
+		Create(context.TODO(), &secret, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// BootstrapTykPortalSecret creates a secret required by Tyk Developer Portal pod which
+// is not going to be ready until this secret is created. If there is a secret created already,
+// it deletes the existing one and recreates the secret.
+func (c *Client) BootstrapTykPortalSecret() error {
```

Comment:
Why don't we move this function to `portal.go` file?

Reply:
that makes sense 👍 i'll move it to there